analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.R

```diff
@@ -224,13 +225,28 @@ mutation_fill_df <- function(sample, ref_df, col_name, default = 0){
   }
 }
 
-
+mutation_gene <- mutations %>% filter(Hugo_Symbol == "NF2")
 EPN_notebook[,"NF2_variant_HGVSc"] <- unlist(sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA,
-                                          function(x) mutation_fill_df(sample = x, ref_df = mutations, col_name = 'HGVSc')))
+                                          function(x) mutation_fill_df(sample = x, ref_df = mutation_gene, col_name = 'HGVSc')))
 EPN_notebook[,"NF2_variant_consequence"] <- unlist(sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA,
-                                          function(x) mutation_fill_df(sample = x, ref_df = mutations, col_name = 'Consequence')))
+                                          function(x) mutation_fill_df(sample = x, ref_df = mutation_gene, col_name = 'Consequence')))
 EPN_notebook[,"NF2_variant_classification"] <- unlist(sapply(EPN_notebook$Kids_First_Biospecimen_ID_DNA,
-                                                   function(x) mutation_fill_df(sample = x, ref_df = mutations, col_name = 'Variant_Classification')))
+                                                   function(x) mutation_fill_df(sample = x, ref_df = mutation_gene, col_name = 'Variant_Classification')))
+mutation_gene <- mutations %>% filter(Hugo_Symbol == "H3F3A")
```

Comment:
```suggestion
mutation_gene <- mutations %>% filter(Hugo_Symbol == "H3F3A" & HGVSp_Short == 'p.K28M')
```

The code runs fine now as is, but I'm suggesting this because there may be samples in the future that have multiple mutations in these H3 genes, which would cause issues when trying to add these values to the data frame in the next line. This is currently the case with non-EPN samples. This would exclude any non-p.K28M mutations from the data frame, but not sure they are necessary.

Reply:
@rjcorb Thank you for reviewing. In the next script `03-summary.Rmd` the subtyping is done and for each patient each column for the 5 histone genes is checked for the p.K28M mutation being present. So it will not matter if other mutations are pulled into the `mutation_gene` dataframe because the requirement for subtyping was only the presence of that `p.K28M` mutation irrespective of other mutations being present. Unless we want to change the requirement that only that mutation should be present. 
CC: @jharenza 

 https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/6e1ac669da58723d6a447eb1cbe1af4a8a11cef9/analyses/molecular-subtyping-EPN/03-summary.Rmd#L160-L166

Reply:
@atlas4213 I agree that this doesn't cause any issues with subtyping. But, in the future, a sample may have multiple different mutations in one of these genes, and if this is the case they would all be pulled into the `mutation_gene` data frame. This would cause an error when trying to add the mutation to the `EPN_notebook` since the `mutation_fill_df` function expects a single value.